spec/models/event_stream_spec.rb

```diff
@@ -0,0 +1,28 @@
+def with_event_types
+  EventStream.event_groups.each do |group_name, group_data|
+    EventStream::GROUP_LEVELS.each do |level|
+      group_data[level]&.each { |typ| yield(group_name, level, typ) }
+    end
+  end
+end
+
+describe EventStream do
+  describe ".event_groups" do
+    with_event_types do |group, level, typ|
+      it ":#{group}/:#{level}/#{typ} is string or regex" do
+        expect(typ.kind_of?(Regexp) || typ.kind_of?(String)).to eq(true)
+      end
+
+      if typ.kind_of?(Regexp)
+        it ":#{group}/:#{level}/#{typ} is usable in SQL queries" do
+          expect { EventStream.where("event_type ~ ?", typ.source).to_a }
+            .to_not raise_error
```

Comment:
I think the only issue here is that `.to_not raise_error` requires a specific exception, but not 100% sure.